packages/grafana-runtime/src/utils/DataSourceWithBackend.ts

```diff
@@ -219,14 +219,21 @@ class DataSourceWithBackend<
    * Make a GET request to the datasource resource path
    */
   async getResource(path: string, params?: any): Promise<any> {
-    return getBackendSrv().get(`/api/datasources/${this.id}/resources/${path}`, params);
+    return getBackendSrv().get(this.getResourcePath(path), params);
   }
 
   /**
    * Send a POST request to the datasource resource path
    */
   async postResource(path: string, body?: any): Promise<any> {
-    return getBackendSrv().post(`/api/datasources/${this.id}/resources/${path}`, { ...body });
+    return getBackendSrv().post(this.getResourcePath(path), { ...body });
+  }
+
+  /**
+   * Get the datasource resource path for the given path
+   */
+  getResourcePath(path: string): string {
+    return `/api/datasources/${this.id}/resources/${path}`;
   }
```

Comment:
The `getResourcePath` method was separated out so that it could be used by `areExamplarsAvailable`, instead of `getResource`. It uses the path to make a `fetch`, using error suppressing options.